opentelemetry-resource-detectors/src/k8stest.rs

```diff
@@ -0,0 +1,43 @@
+#[cfg(test)]
+mod tests {
+    use super::*;
+    use std::env;
+
+    #[test]
+    fn test_k8s_resource_detector_with_env_vars() {
+        // Set environment variables for testing
+        env::set_var("K8S_POD_NAME", "test-pod");
```

Comment:
better to use [temp-env](https://crates.io/crates/temp-env) package for testing with env-varaibles, will prevent conflict in case of concurrent test runs. Eg - https://github.com/open-telemetry/opentelemetry-rust/blob/ea4b5e4d992610bb23a059f05253d0c47e262015/opentelemetry-sdk/src/resource/env.rs#L102

Reply:
Oh, thanks for the feedback @lalitb Done now, kindly review and let me know if there are any further changes